api/app/src/mappings/apple/index.ts

```diff
@@ -0,0 +1,158 @@
+
+  return payload
+}
+
+export const hasActivity = (data: AppleHealth): Boolean => {
+  return !!data.HKQuantityTypeIdentifierActiveEnergyBurned ||
+    !!data.HKQuantityTypeIdentifierStepCount ||
+    !!data.HKQuantityTypeIdentifierActiveEnergyBurned ||
+    !!data.HKQuantityTypeIdentifierBasalEnergyBurned ||
+    !!data.HKQuantityTypeIdentifierFlightsClimbed
+};
+
+export const hasBody = (data: AppleHealth): Boolean => {
+  return !!data.HKQuantityTypeIdentifierHeight ||
+    !!data.HKQuantityTypeIdentifierLeanBodyMass ||
+    !!data.HKQuantityTypeIdentifierBodyMass ||
+    !!data.HKQuantityTypeIdentifierBodyFatPercentage
+};
+
+export const hasNutrition = (data: AppleHealth): Boolean => {
+  return !!data.HKQuantityTypeIdentifierDietaryCaffeine ||
+    !!data.HKQuantityTypeIdentifierDietaryCalcium ||
+    !!data.HKQuantityTypeIdentifierDietaryCarbohydrates ||
+    !!data.HKQuantityTypeIdentifierDietaryCholesterol ||
+    !!data.HKQuantityTypeIdentifierDietaryCopper ||
+    !!data.HKQuantityTypeIdentifierDietaryEnergyConsumed ||
+    !!data.HKQuantityTypeIdentifierDietaryFatTotal ||
+    !!data.HKQuantityTypeIdentifierDietaryFiber ||
+    !!data.HKQuantityTypeIdentifierDietaryFolate ||
+    !!data.HKQuantityTypeIdentifierDietaryIodine ||
+    !!data.HKQuantityTypeIdentifierDietaryMagnesium ||
+    !!data.HKQuantityTypeIdentifierDietaryManganese ||
+    !!data.HKQuantityTypeIdentifierDietaryNiacin ||
+    !!data.HKQuantityTypeIdentifierDietaryPantothenicAcid ||
+    !!data.HKQuantityTypeIdentifierDietaryPhosphorus ||
+    !!data.HKQuantityTypeIdentifierDietaryPotassium ||
+    !!data.HKQuantityTypeIdentifierDietaryProtein ||
+    !!data.HKQuantityTypeIdentifierDietaryRiboflavin ||
+    !!data.HKQuantityTypeIdentifierDietarySelenium ||
+    !!data.HKQuantityTypeIdentifierDietarySodium ||
+    !!data.HKQuantityTypeIdentifierDietarySugar ||
+    !!data.HKQuantityTypeIdentifierDietaryThiamin ||
+    !!data.HKQuantityTypeIdentifierDietaryVitaminA ||
+    !!data.HKQuantityTypeIdentifierDietaryVitaminB6 ||
+    !!data.HKQuantityTypeIdentifierDietaryVitaminB12 ||
+    !!data.HKQuantityTypeIdentifierDietaryVitaminC ||
+    !!data.HKQuantityTypeIdentifierDietaryVitaminD ||
+    !!data.HKQuantityTypeIdentifierDietaryVitaminE ||
+    !!data.HKQuantityTypeIdentifierDietaryVitaminK ||
+    !!data.HKQuantityTypeIdentifierDietaryWater ||
+    !!data.HKQuantityTypeIdentifierDietaryZinc
+}
+
+export const hasBiometrics = (data: AppleHealth): Boolean => {
+  return !!data.HKQuantityTypeIdentifierHeartRate ||
+    !!data.HKQuantityTypeIdentifierRestingHeartRate ||
+    !!data.HKQuantityTypeIdentifierHeartRateVariabilitySDNN ||
+    !!data.HKQuantityTypeIdentifierBodyTemperature ||
+    !!data.HKQuantityTypeIdentifierBloodPressureSystolic ||
+    !!data.HKQuantityTypeIdentifierBloodPressureDiastolic ||
+    !!data.HKQuantityTypeIdentifierRespiratoryRate
+};
+
+
+export const appleItem = z.object({ date: z.string(), value: z.number() })
```

Comment:
We could check the date is in the format specified by Apple and is a valid value.

For reference: https://github.com/metriport/metriport/blob/develop/api/app/src/mappings/garmin/index.ts#L31

Reply:
So i did this 
  date: z.string().refine((v) => dayjs(v, "YYYY-MM-DDTHH:mm:ss", true).isValid())
  
  No matter what i did though it always returns false but in a repl it works just fine its weird.